kyaml/kio/filters/merge3.go

```diff
@@ -114,8 +114,9 @@ type tuples struct {
 	mergeOnPath bool
 }
 
-// isSameResource returns true if meta1 and meta2 are for the same logic resource
-func (ts *tuples) isSameResource(meta1, meta2 yaml.ResourceMeta) bool {
+// IsSameResource returns true if meta1 and meta2 are for the same logical resource
+// this implementation can be overridden with other implementation
+var IsSameResource = func(meta1, meta2 yaml.ResourceMeta, mergeOnPath bool) bool {
```

Comment:
I think making this extensible by allowing users to provide an implementation of an interface (something like `ResourceMatcher`?) on the `Merge3` struct would be a better solution here. It would then be defaulted to the current implementation if the user doesn't provide one.
Also, the `mergeOnPath` variable seems like something that should be tuned by providing a specific implementation, rather than making it part of a function on an interface. Not sure what is the best way to do this without breaking existing functionality.

Reply:
Done.